tubular/mapping.py

```diff
@@ -78,8 +95,15 @@ def transform(self, X):
 
         self.check_is_fitted(["mappings"])
 
+        # Store the original dtypes of the columns
+        mapped_columns = self.mappings.keys()
+        original_dtypes = X[mapped_columns].dtypes
+
         X = super().transform(X)
 
+        # Call the check_dtype_changes method
+        self.check_dtype_changes(X, original_dtypes, suppress_dtype_warning)
```

Comment:
I'm not sure we need the check here as super().transform() doesn't do any mappings?